src/components/App/App.jsx

```diff
@@ -0,0 +1,116 @@
+import Navbar from "../Navbar/Navbar"
+import Home from "../Home/Home"
+import Profile from "../Profile/Profile"
+import Recipes from "../Recipes/Recipes"
+import Pantry from "../Pantry/Pantry"
+import ShoppingCart from "../ShoppingCart/ShoppingCart"
+import './App.css';
+
+import { BrowserRouter, Routes, Route } from "react-router-dom"
+import { useState, useEffect } from "react"
+
+export default function App() {
+
+  var basicProducts = [
+    {name: "apple",
+    quantity: 2},
+    {name: "orange",
+    quantity: 5},
+    {name: "grape",
+    quantity: 8},
+    {name: "bacon",
+    quantity: 3},
+    {name: "milk",
+    quantity: 2},
+    {name: "pasta",
+    quantity: 1}
+  ]
+
+  var basicProductForm = {
+    name: "",
+    quantity: ""
+  }
+
+const [products, setProducts] = useState(basicProducts)
+const [productForm, setProductForm] = useState(basicProductForm)
+
+const handleProductQuantity = (productName, operation) => {
+  // find index of productName
+  let itemIndex = products.findIndex(item => item.name === productName)
+
+  let newProducts = []
+  for (let i = 0; i < products.length; i++) {
+    newProducts.push(products[i])
+  }
+
+  if (operation == "add") {
+    newProducts[itemIndex].quantity += 1
```

Comment:
Is it possible that the item wasn't found (`itemIndex===-1`)? If so this could cause an error.

Reply:
It wouldn't be possible since the buttons on the product would only be clicked if that product exists in the first place, but adding an error would be safer. I will add a TODO comment here so I can add this safety feature later on.